mempool/priority/mempool.go

```diff
@@ -0,0 +1,809 @@
+package v1
```

Comment:
[nit] thanks for renaming the directory from v1 to priority because priority is clearer. Can we also rename the package so that we obey the loose Go convention of naming the package the same thing as the directory? Could be a follow-up.

```suggestion
package priority
```